### PR TITLE
docs: Monitoring doc for Kubernetes

### LINKED ISSUE #1331: Monitoring and alerting in Kubernetes
### Describe the Issue

We need to provide Kubernetes users with guidance on setting up monitoring and alerting for their clusters. We should at least provide an equivalent topic to https://docs.redpanda.com/docs/manage/monitoring/ that has example instructions for setting up Prometheus and Grafana in Kubernetes.

### New feature or content gap requests

Content gap.

### PR DESCRIPTION
**Pull request by JakeSCahill**
Review deadline: March 17

Preview: https://deploy-preview-1397--redpanda-documentation.netlify.app/docs/manage/kubernetes/monitor/#configure-prometheus

Fixes #1331 

This PR single-sources all the information from the existing monitoring doc and adds Kubernetes-specific instructions for configuring Prometheus.

Also added context switchers for topics that have a Kubernetes equivalent.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204218351694941

### CODE REVIEW

**Review by RafalKorepta — APPROVED** [review]
LGTM

**Comment by micheleRP** [comment]
@JakeSCahill will you please add the new Monitor page here: https://deploy-preview-1397--redpanda-documentation.netlify.app/docs/manage/kubernetes/? And while you're editing that file, will you pls change Rolling Upgrades to either Upgrade?

**Comment by JakeSCahill** [comment]
@micheleRP I've updates the rolling upgrades in #1406

**Comment by JakeSCahill** [comment]
@micheleRP I've updated the rolling upgrades in #1406

**Comment by JakeSCahill** [comment]
@micheleRP I've updated the rolling upgrades in #1406

File: docs/manage/shared/monitoring/_grafana-dashboard.mdx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import Link from '@docusaurus/Link';
+
+[Grafana](https://grafana.com/oss/grafana/) is a tool to query, visualize, and generate alerts for metrics.
+
+Redpanda supports generating Grafana dashboards from its metrics endpoints with `rpk generate grafana-dashboard`.
+
+To generate a comprehensive Grafana dashboard, run the following command and pipe the output to a file that can be imported into Grafana:
+
+```bash
+rpk generate grafana-dashboard --datasource <name> --metrics-endpoint <url> > <output-file>
```
[the remark this answers is not in the document]
Comment (JakeSCahill) [reply docs/manage/shared/monitoring/_grafana-dashboard.mdx:10]: This PR is mirroring the existing monitoring doc but for Kubernetes. If we have a preferred Grafana dashboard, we can fix it in a separate issue.

File: docs/manage/shared/monitoring/_throughput.mdx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import Link from '@docusaurus/Link';
+
+While maximizing the rate of messages moving from producers to brokers then to consumers depends on tuning each of those components, the total throughput of all topics provides a system-level metric to monitor. Then when you observe abnormal, unhealthy spikes or dips in producer or consumer throughput, look for correlation with changes in the number of active connections (<Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_rpc_active_connections'}>`redpanda_rpc_active_connections`</Link>) and logged errors to drill down to the root cause.
```
Comment (micheleRP) [inline docs/manage/shared/monitoring/_throughput.mdx:3]:
```suggestion
While maximizing the rate of messages moving from producers to brokers then to consumers depends on tuning each of those components, the total throughput of all topics provides a system-level metric to monitor. When you observe abnormal, unhealthy spikes or dips in producer or consumer throughput, look for correlation with changes in the number of active connections (<Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_rpc_active_connections'}>`redpanda_rpc_active_connections`</Link>) and logged errors to drill down to the root cause.
```

File: docs/manage/shared/monitoring/_latency.mdx [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import Link from '@docusaurus/Link';
+
+Latency should be consistent between produce and fetch sides. It should also be consistent over time. Take periodic snapshots of produce and fetch latencies, including at upper percentiles (95%, 99%), and watch out for significant changes over a short duration.
+
+In Redpanda the latency of produce and fetch requests includes the latency of internal inter-broker RPCs that are born from Redpanda's internal implementation using Raft.
```
Comment (micheleRP) [inline docs/manage/shared/monitoring/_latency.mdx:5]:
```suggestion
In Redpanda, the latency of produce and fetch requests includes the latency of inter-broker RPCs that are born from Redpanda's internal implementation using Raft.
```

File: docs/manage/shared/monitoring/_latency.mdx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+#### Kafka consumer latency
+
+To monitor Kafka consumer request latency, use the <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_kafka_request_latency_seconds'}>`redpanda_kafka_request_latency_seconds`</Link> histogram with the label `redpanda_request="consume"`. For example, create a query for the 99th percentile:
+
+```
+histogram_quantile(0.99, sum(rate(redpanda_kafka_request_latency_seconds_bucket{redpanda_request="consume"}[5m])) by (le, provider, region, instance, namespace, pod))
+```
+
+You can monitor the rate of Kafka consumer requests using`redpanda_kafka_request_latency_seconds_count` with the `redpanda_request="consume"` label:
```
Comment (micheleRP) [inline docs/manage/shared/monitoring/_latency.mdx:15]:
```suggestion
You can monitor the rate of Kafka consumer requests using `redpanda_kafka_request_latency_seconds_count` with the `redpanda_request="consume"` label:
```

File: docs/manage/shared/monitoring/_partition-health.mdx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+A healthy cluster has partition data fully replicated across its brokers. 
+
+An under-replicated partition is at higher risk of data loss. It also adds latency because messages must be replicated before being committed. To know when a partition isn't fully replicated, create an alert for the <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_kafka_under_replicated_replicas'}>`redpanda_kafka_under_replicated_replicas`</Link> gauge when it is greater than zero:
+
+```
+redpanda_kafka_under_replicated_replicas > 0
+```
+
+Under-replication can be caused by unresponsive brokers. When an alert on `redpanda_kafka_under_replicated_replicas` is triggered, identify the problem broker(s) and examine their log(s).
```
Comment (micheleRP) [inline docs/manage/shared/monitoring/_partition-health.mdx:27]:
```suggestion
Under-replication can be caused by unresponsive brokers. When an alert on `redpanda_kafka_under_replicated_replicas` is triggered, identify the problem broker(s) and examine its log. 
```

File: docs/manage/shared/monitoring/_partition-health.mdx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+A healthy cluster has a leader for every partition. 
+
+A partition without a leader cannot exchange messages with producers or consumers. To identify when a partition doesn't have a leader, create an alert for the <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_cluster_unavailable_partitions'}>`redpanda_cluster_unavailable_partitions`</Link> gauge when it is greater than zero: 
+
+```
+redpanda_cluster_unavailable_partitions > 0
+```
+
+Leaderless partitions can be caused by unresponsive brokers. When an alert on `redpanda_cluster_unavailable_partitions` is triggered, identify the problem broker(s) and examine their log(s).
```
Comment (micheleRP) [inline docs/manage/shared/monitoring/_partition-health.mdx:39]:
```suggestion
Leaderless partitions can be caused by unresponsive brokers. When an alert on `redpanda_cluster_unavailable_partitions` is triggered, identify the problem broker(s) and examine its log. 
```

File: docs/manage/shared/monitoring/_partition-health.mdx [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+```
+redpanda_cluster_unavailable_partitions > 0
+```
+
+Leaderless partitions can be caused by unresponsive brokers. When an alert on `redpanda_cluster_unavailable_partitions` is triggered, identify the problem broker(s) and examine their log(s). 
+
+#### Raft RPCs
+
+Redpanda's Raft implementation exchanges periodic status RPCs between a broker and its peers. The <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_node_status_rpcs_timed_out'}>`redpanda_node_status_rpcs_timed_out`</Link> gauge increases when a status RPC times out for a peer, which indicates a peer may be unresponsive and may lead to problems with partition replication that Raft manages. Monitor for non-zero values of this gauge, and correlate it with any logged errors or changes partition replication.
```
Comment (micheleRP) [inline docs/manage/shared/monitoring/_partition-health.mdx:43]:
```suggestion
Redpanda's Raft implementation exchanges periodic status RPCs between a broker and its peers. The <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_node_status_rpcs_timed_out'}>`redpanda_node_status_rpcs_timed_out`</Link> gauge increases when a status RPC times out for a peer, which indicates that a peer may be unresponsive and may lead to problems with partition replication that Raft manages. Monitor for non-zero values of this gauge, and correlate it with any logged errors or changes partition replication.
```

Comment (micheleRP) [inline docs/manage/shared/monitoring/_partition-health.mdx:43]:
```suggestion
Redpanda's Raft implementation exchanges periodic status RPCs between a broker and its peers. The <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_node_status_rpcs_timed_out'}>`redpanda_node_status_rpcs_timed_out`</Link> gauge increases when a status RPC times out for a peer, which indicates that a peer may be unresponsive and may lead to problems with partition replication that Raft manages. Monitor for non-zero values of this gauge, and correlate it with any logged errors or changes partition replication.
```

File: docs/manage/shared/monitoring/_consumers.mdx [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import Link from '@docusaurus/Link';
+
+#### Consumer group lag
+
+When working with Kafka consumer groups, the consumer group lag&mdash;the difference between the broker's latest (max) offset and the group's last committed offset&mdash;is a performance indicator of how fresh the data being consumed is. While higher lag for archival consumers is expected, high lag for real-time consumers could indicate that the consumers are overloaded and thus may need their topics to be partitioned more, or to spread the load to more consumers.   
+
+To monitor consumer group lag, create a query with the <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_kafka_max_offset'}>`redpanda_kafka_max_offset`</Link> and <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_kafka_consumer_group_committed_offset'}>`redpanda_kafka_consumer_group_committed_offset`</Link> gauges:
```
Comment (micheleRP) [inline docs/manage/shared/monitoring/_consumers.mdx:7]: `redpanda_kafka_max_offset` renders with backticks

File: docs/manage/shared/monitoring/_services.mdx [hunk 9]
```diff
@@ -0,0 +1,41 @@
+Monitor the health of specific Redpanda services with the following metrics. 
+
+#### Schema registry
```
Comment (micheleRP) [inline docs/manage/shared/monitoring/_services.mdx:3]: I think this should be capitalized like below (Schema Registry)

**Review by micheleRP — APPROVED** [review]
some little things to fix, but looks good!

File: docs/manage/kubernetes/monitor.mdx [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+  ```
+
+1. Expose the Prometheus server to your localhost:
+
+  ```bash
+  export POD_NAME=$(kubectl get pods --namespace monitoring -l "app=prometheus,component=server" -o jsonpath="{.items[0].metadata.name}")
+  kubectl --namespace monitoring port-forward $POD_NAME 9090
+  ```
+
+1. [Open Prometheus](http://localhost:9090/graph), and see that metrics from Redpanda endpoints are scraped.
```
[the remark this answers is not in the document]
Comment (JakeSCahill) [reply docs/manage/kubernetes/monitor.mdx:81]: that's where prometheus is hosted when you run the command above.

File: docs/manage/shared/monitoring/_consumers.mdx (hunk 8 above)
Comment (JakeSCahill) [inline docs/manage/shared/monitoring/_consumers.mdx:7]:
```suggestion
To monitor consumer group lag, create a query with the <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_kafka_max_offset'}><code>redpanda_kafka_max_offset</code></Link> and <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_kafka_consumer_group_committed_offset'}><code>redpanda_kafka_consumer_group_committed_offset</code></Link> gauges:
```

File: docs/manage/shared/monitoring/_infrastructure-resources.mdx [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+#### CPU usage
+
+For the total CPU uptime, monitor <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_uptime_seconds_total'}>`redpanda_uptime_seconds_total`</Link>. Monitoring its rate of change with the following query can help detect unexpected dips in uptime:
+
+```
+rate(redpanda_uptime_seconds_total[5m])
+```
+
+For the total CPU busy (non-idle) time, monitor <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_cpu_busy_seconds_total'}>`redpanda_cpu_busy_seconds_total`</Link>. Monitoring its rate of change, the percentage a CPU core ("shard") is in use at a given point in time, with the following query can help detect unexpected idling:
```
Comment (JakeSCahill) [inline docs/manage/shared/monitoring/_infrastructure-resources.mdx:13]: How about:

```suggestion
For the total CPU busy (non-idle) time, monitor <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_cpu_busy_seconds_total'}>`redpanda_cpu_busy_seconds_total`</Link>.

To detect unexpected idling, you can query the rate of change as a percentage of the CPU core ("shard") that is in use at a given point in time.
```

cc @shorgi @micheleRP

File: docs/manage/shared/monitoring/_infrastructure-resources.mdx [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+
+```
+rate(redpanda_cpu_busy_seconds_total[5m])
+```
+
+#### Memory allocated
+
+To monitor the percentage of memory allocated, use a formula with <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_memory_allocated_memory'}>`redpanda_memory_allocated_memory`</Link> and <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_memory_free_memory'}>`redpanda_memory_free_memory`</Link>:
```
Comment (JakeSCahill) [inline docs/manage/shared/monitoring/_infrastructure-resources.mdx:21]:
```suggestion
To monitor the percentage of memory allocated, use a formula with <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_memory_allocated_memory'}><code>redpanda_memory_allocated_memory</code></Link> and <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_memory_free_memory'}><code>redpanda_memory_free_memory</code></Link>:
```

File: docs/manage/shared/monitoring/_infrastructure-resources.mdx [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+
+```
+sum(redpanda_memory_allocated_memory) / (sum(redpanda_memory_free_memory) + sum(redpanda_memory_allocated_memory))
+```
+
+#### Disk used
+
+To monitor the percentage of disk consumed, use a formula with <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_storage_disk_free_bytes'}>`redpanda_storage_disk_free_bytes`</Link> and <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_storage_disk_total_bytes'}>`redpanda_storage_disk_total_bytes`</Link>:
```
Comment (JakeSCahill) [inline docs/manage/shared/monitoring/_infrastructure-resources.mdx:29]:
```suggestion
To monitor the percentage of disk consumed, use a formula with <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_storage_disk_free_bytes'}><code>redpanda_storage_disk_free_bytes</code></Link> and <Link to={props.frontmatter && props.frontmatter.linkRoot + 'reference/public-metrics-reference#redpanda_storage_disk_total_bytes'}><code>redpanda_storage_disk_total_bytes</code></Link>: 
```

File: docs/manage/shared/monitoring/_services.mdx (hunk 9 above)
Comment (JakeSCahill) [inline docs/manage/shared/monitoring/_services.mdx:3]:
```suggestion
#### Schema Registry
```

File: docs/manage/shared/monitoring/_infrastructure-resources.mdx (hunk 11 above)
[the remark this answers is not in the document]
Comment (JakeSCahill) [reply docs/manage/shared/monitoring/_infrastructure-resources.mdx:13]: See suggestion below

**Pull request merged**